packages/shared/src/store/sync.ts

```diff
@@ -598,11 +596,24 @@ export const clearSyncQueue = () => {
   syncQueue.clear();
 };
 
-export const start = async () => {
-  api.subscribeUnreads(handleUnreadUpdate);
-  api.subscribeGroups(handleGroupUpdate);
-  api.subscribeToChannelsUpdates(handleChannelsUpdate);
-  api.subscribeToChatUpdates(handleChatUpdate);
-  api.subscribeToContactUpdates(handleContactUpdate);
-  useStorage.getState().start();
+export const initializeStorage = () => {
+  return syncQueue.add('initialize storage', async () => {
+    return useStorage.getState().start();
+  });
+};
+
+export const setupSubscriptions = async () => {
+  return syncQueue.add('setup subscriptions', async () => {
+    await Promise.all([
```

Comment:
maybe a comment as to why you're making this load order explicit

Reply:
Going to remove as I don't remember why and it doesn't seem to have much of an effect.